test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -498,13 +498,12 @@ def forward(self, x, y):
                 # view + linear + view(joint_graph+freeze pass)
                 match_count = match_count + 5 if is_inplace else match_count + 3
                 match_nodes = match_nodes + 7 if is_inplace else match_nodes + 5
-            mod = M(binary_fn, input_shape[-1], out_feature, bias).to(dtype).eval()
-            v = torch.randn(input_shape).to(dtype)
+            mod = M(binary_fn, input_shape[-1], out_feature, bias).eval()
+            v = torch.randn(input_shape)
             other = torch.randn(input_shape[:-1] + [out_feature]).to(dtype)
```

Comment:
Why do we not convert dtype on `mod` and input `v` but convert it on "out" here?

Reply:
Hi, Jiong.
Here we choose do not convert dtype on `mod` and `v` because we expected autocast to handle it. And for `other`, the autocast will not cast it because `add` is full-through op.
```
y = linear(x) + z
``` 
And currently we do not fuse "+ z" if z is float and linear(x) is lp.